### ISSUE TITLE
SAE for Instruct Models

### ISSUE THREAD

**Issue by LehengTHU** [OpenMOSS/Language-Model-SAEs#81]
Hi, Llama Scope teams. Thanks for your wonderful work! 
Do you have any plans for releasing SAEs trained on instruct-finetuned models? I notice the loss at the last layer is much bigger on instruct-finetuned models (Section 4.5.2). I’m curious whether this suggests that these SAEs cannot be directly applied to instruct-finetuned models for steering, as it might lead to unexpected issues.
Thanks for your time!
Best.

**Comment by Hzfinfdu** [comment]
Hi, thanks for this question and also apologies for the late response.

Actually we did not train SAEs on instruct models. But it is indeed reasonable to try to finetune llamascope SAEs to instruct models.

Section 4.5.2 implies that most SAEs can be transferred across base and instruct models, with the exception of the last layer. Personally I do not think this will lead to the conclusion that these SAEs are not suitable for steering. Most concepts that may be of interest should lie in the middle layers (i.e. L10-20). I believe a common practice of model steering is applying steering vectors here, instead of the last layer. And that should instantly work with LlamaScope SAEs applied to instruct model residual streams. It is advised to find desired features to steer at [neuronpedia.org](https://neuronpedia.org) .

Nonetheless, we think the exception at the last layer is an interesting phenomenon. We are not sure whether this is a common issue or something we created by mistake. If it's the former case, it can be interesting to see what is actually going on at the last layer.

**Comment by LehengTHU** [comment]
Thanks for your detailed reply! 

It helps me a lot to understand your great works better. I will try to adopt these SAEs on instruct models following your suggestions. Yeah, I noticed, except for the last layer, that these SAEs work fine on instruct models. I also agree with your idea that the last layer is a very interesting topic of interest.
BTW, may I kindly ask whether your teams plan to train SAEs on instruct-finetuned models in the future?

Best.

**Comment by LehengTHU** [comment]
Got that! Thank you so much!

Wish happy CNY.
Best.
